stack/app/nat.tf

```diff
@@ -1,6 +1,7 @@
 module "nat" {
-  source = "../../nat"
+  count = length(var.public_ips) > 0 ? 1 : 0
```

Comment:
this is a breaking change. If there wasn't a public IP before, we would still create route tables. Are we sure, no app depends on having those route tables?